Unity/Assets/Scripts/MyTouchObserver.cs

```diff
@@ -1,123 +1,126 @@
+			get { return m_Transform = m_Transform ?? GetComponent<Transform>(); }
+		}
 
-        public List<Touch> Touches {
-            get { return m_Touches = m_Touches ?? new List<Touch>(); }
-        }
+		public List<Touch> Touches {
+			get { return m_Touches = m_Touches ?? new List<Touch>(); }
+		}
 
 		#endregion
 
 
 		#region TouchHandlers
 
 		public bool OnTouchBegan(Touch touch) {
-            if (HitTest(touch)) {
-                if (handleMultipleTouches) {
-                    this.Touches.Add(touch);
-                    ReactTouchIn();
-                    return true;
-                }
-                if (m_Touch == null) {
-                    m_Touch = touch;
-                    ReactTouchIn();
-                    return true;
-                }
-            }
-            return false;
-        }
-
-        public void OnTouchCancelled(Touch touch) {
-            OnTouchEnded(touch);
-        }
-
-        public void OnTouchEnded(Touch touch) {
-            ReactTouchOut();
-            if (handleMultipleTouches) {
-                this.Touches.Remove(touch);
-            }
-            else {
-                m_Touch = null;
-            }
-        }
-
-        public void OnTouchMoved(Touch touch) {
-        }
-
-        #endregion
-
-
-        #region Methods
-
-        bool HitTest(Touch touch) {
-            Bounds bounds = this.Renderer.bounds;
-            bounds.extents += Vector3.forward;
-            return bounds.Contains(CameraUtils.TouchToWorldPoint(touch, Transform, Camera));
-        }
-
-        void ReactTouchIn() {
-            this.Transform.localScale = Vector3.one * 1.2f;
-        }
-
-        void ReactTouchOut() {
-            this.Transform.localScale = Vector3.one;
-        }
-
-        void OnEnable() {
-            if (TouchDispatcher.Instance) {
-                TouchDispatcher.Instance.Add(this, 0, true);
-            }
-        }
-
-        void OnDisable() {
-            if (TouchDispatcher.Instance) {
-                TouchDispatcher.Instance.Remove(this);
-            }
-        }
-
-        #endregion
-
-    }
+			if (HitTest(touch)) {
+				if (handleMultipleTouches) {
+					this.Touches.Add(touch);
+					ReactTouchIn();
+					return true;
+				}
+				if (m_Touch == null) {
+					m_Touch = touch;
+					ReactTouchIn();
+					return true;
+				}
+			}
+			return false;
+		}
+
+		public void OnTouchCancelled(Touch touch) {
+			OnTouchEnded(touch);
+		}
+
+		public void OnTouchEnded(Touch touch) {
+			if (handleMultipleTouches) {
+				this.Touches.Remove(touch);
+				if (this.Touches.Count < 1) {
+					ReactTouchOut();
+				}
+			}
+			else {
+				ReactTouchOut();
+				m_Touch = null;
+			}
+		}
+
+		public void OnTouchMoved(Touch touch) {
+		}
+
+		#endregion
+
+
+		#region Methods
+
+		private bool HitTest(Touch touch) {
+			Bounds bounds = this.Renderer.bounds;
+			bounds.extents += Vector3.forward;
+			return bounds.Contains(CameraUtils.TouchToWorldPoint(touch, Transform, Camera));
```

Comment:
When referencing a class' own properties, include `this`.  This helps distinguish between, for example, a static property.  e.g.
`return bounds.Contains(CameraUtils.TouchToWorldPoint(touch, this.Transform, this.Camera));`